### PR TITLE
chore: change headless-demo from xml to json

### LINKED ISSUE #1335: Update the headless demo to use JSON instead of XML
**Category**

 - Codelabs

**Component**

headless-demo

**Describe the bug**

XML is iceboxed, so our demos should use JSON.

See: https://google.github.io/blockly-samples/examples/headless-demo/

**How to fix**
In `examples/headless-demo/index.html` you want to:
1) Change the contents of the text area to be valid JSON instead of XML. You can get this by
  a) deserializing the existing XML into the [playground](https://blockly-demo.appspot.com/static/tests/playground.html) using the text area on the left, and the "Load" button
  b) Hitting the "Save JSON" button.
  c) Pasting the result back into `index.html`
2) Change the call to [Blockly.Xml.domToWorkspace](https://developers.google.com/blockly/reference/js/blockly.xml_namespace.domtoworkspace_1_function.md) to a call to [Blockly.serialization.workspaces.load](https://developers.google.com/blockly/reference/js/blockly.serialization_namespace.workspaces_namespace.load_1_function.md).

### PR DESCRIPTION
**Pull request by mikebarr24**
closes #1335

### CODE REVIEW

**Comment by mikebarr24** [comment]
Trying to change commit naming

File: examples/headless-demo/index.html [hunk 1]
```diff
@@ -26,61 +26,117 @@
...
-  </block>
-</xml>
+        <textarea id="json_input">
+          {
+            "blocks": {
+              "languageVersion": 0,
+              "blocks": [
+                {
+                  "type": "controls_if",
+                  "id": "@Tr`@6~sR~{^SM209=Xf",
```
Comment (BeksOmega) [inline examples/headless-demo/index.html]: Could you remove any references to the 'id' property? These are unnecessary in this situation =)

File: examples/headless-demo/index.html [hunk 2]
```diff
@@ -97,18 +153,19 @@
...
-      var xmlText = document.getElementById('xml_input').value;
+      // Parse the JSON into a tree.
+      var jsonText = document.getElementById('json_input').value;
+      var code;
       try {
-        var xml = Blockly.Xml.textToDom(xmlText)
+        var json = JSON.parse(jsonText)
+        var demoWorkspace = new Blockly.Workspace();
+        Blockly.serialization.workspaces.load(json, demoWorkspace);
+        code = Blockly.Python.workspaceToCode(demoWorkspace);
```
Comment (BeksOmega) [inline examples/headless-demo/index.html]: I don't think these have the ability to throw. Did you run into issues, or just moving these to be extra cautious?

File: examples/headless-demo/package.json [hunk 3]
```diff
@@ -13,6 +13,6 @@
     ]
   },
   "dependencies": {
-    "blockly": ">=5.20210325.0 <9"
+    "blockly": "^9.0.0"
```
Comment (BeksOmega) [inline examples/headless-demo/package.json:16]: Yay ty for the update!

**Review by BeksOmega — CHANGES_REQUESTED** [review]
Looks great! Just a few little fixes =)

Also checking: you ran the page, tested it, and everything works correctly?

Once I get fixes + testing confirmation I'll go ahead and merge =)

File: examples/headless-demo/index.html (hunk 1 above)
Comment (mikebarr24) [reply examples/headless-demo/index.html]: Cool! I've removed all refs to 'id'

File: examples/headless-demo/index.html (hunk 2 above)
Comment (mikebarr24) [reply examples/headless-demo/index.html]: I wasn't sure. I've left the `JSON.parse()` in.

**Comment by mikebarr24** [comment]
@BeksOmega I've tested this in so much as I've run the json which was generated from playground through it. Hope this is OK.

**Review by BeksOmega — APPROVED** [review]

**Comment by BeksOmega** [comment]
> @BeksOmega I've tested this in so much as I've run the json which was generated from playground through it. Hope this is OK.

Yep that's all I was curious about! Just wanted to make sure it would actually generate the code and whatnot hehe.

Thank you for your work on this! We really appreciate it =)

**Pull request merged**